server/controllers/oauth.js

```diff
@@ -0,0 +1,45 @@
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+const generator = require('generate-password');
+
+// @route GET /login/register/user
+// @desc Login if user exists else Register user
+// @access Private
+exports.loginRegisterUser = asyncHandler(async (req, res, next) => {
+
+    // check if the user already exists.
+    let user = await User.findOne({ email: req.user.email })
```

Comment:
`req.user` implies a user is already signed in, which means they already have an account.